core/src/main/scala/com/github/johnynek/paiges/Doc.scala

```diff
@@ -619,6 +619,12 @@ object Doc {
     }
 
   /**
+   * Build a document from a single character.
+   */
+  def char(c: Char): Doc =
+    if (c == '\n') Doc.line else Text(new String(Array(c)))
```

Comment:
we actually have a table of printable ascii characters to reduce allocation. Do you want to use that?